### DISCUSSION TITLE
[v5] updateOrCreateMany not available on related model

### DISCUSSION THREAD

**Discussion by bitkidd** [adonisjs/core#1799] [Help]
Hey!
The method `updateOrCreateMany` is not available on related model in a `hasMany` relation, for example `Product` has many `Variants`, in this situation will be very useful to have it.
Am I missing something?

**Comment by thetutlage** [comment]
Nope, the method doesn't exists. Happy to accept a PR for the same

**Reply by bitkidd** [reply]
I just dived into the code and started adding new method, but then discovered a situation when `updateOrCreateMany` may not work properly, for example when a `Product` has many `Variants` and they do not rely on any of the unique properties, so the only unique column is a primary column, the `id`. When submitting data to the method we do something like this:
```javascript
await product.related('variants').updateOrCreateMany('id', [
  { id: 1, size: 'XS', color: 'Red' }, // Should be updated
  { id: 2, size: 'S', color: 'Blue' }, // Should be updated
  { id: 3, size: 'M', color: 'Green' }, // Should be updated
  { size: 'L', color: 'Red' }, // Should be created
])
```
But following the code this will rise an exception, due to the fact that `id` is `undefined` on the last item, but you can't provide `id` before creation.

The error may be mitigated using a `unique` field, for example a `uid` field with a randomly generated `uuid/v4` or any other random string.

**Reply by thetutlage** [reply]
Following up on the PR https://github.com/adonisjs/lucid/pull/615

**Reply by bitkidd** [reply]
Recreated the PR [adonisjs/lucid#619](https://github.com/adonisjs/lucid/pull/619)

**Comment by leognmotta** [comment]
```js
await product.related('variants').updateOrCreateMany('id', [
  { id: 1, size: 'XS', color: 'Red' }, // Should be updated
  { id: 2, size: 'S', color: 'Blue' }, // Should be updated
  { id: 3, size: 'M', color: 'Green' }, // Should be updated
  { size: 'L', color: 'Red' }, // Should be created
])
```
having this would be nice, but it is still throwing exception
